File: src/FASTCAMPUS/ALGO/D26/P20291.js
const fs = require('fs');
class Scanner {
  constructor(fileName){
    this.read = fs.readFileSync(fileName).toString().split('\n');
    this.stringToken = [];
  }
  next(){
    if(this.stringToken.length === 0){
      if(this.read.length > 0){
        this.stringToken = this.read.shift().split(" ");
      } else {
        return null;
      }
    } 
    return this.stringToken.shift();
  }

  nextNumber(){
    const next = this.next();
    if(next){
      return Number(next);
    } else {
      return null;
    }
    
  }

  nextLine(){
    if(this.read.length > 0){
      return this.read.shift();
    } else {
      return null;
    }
    
  }
}
const scanner = new Scanner('stdin.txt');
let N, a;
const input = function (){
  N = scanner.nextNumber();
  a = [];
  for(let i = 0; i < N; i ++){
    a[i] = scanner.nextLine().split('.')[1];
  }

}

const pro = function (){
  a.sort( (a, b) => {
    let idx = 0;
    while((idx < a.length && idx < b.length) && (a.charAt(idx) === b.charAt(idx)) ){
      idx ++;
    }
    //두 단어의 길이가 같으면
    if(a.length === b.length){
      if(idx == a.length){//같은단어
        return 0;
      } else {
        return a.charCodeAt(idx) - b.charCodeAt(idx);
      }
    } else {
      //a가 b에 포함
      if(idx == a.length){
        return -1;
      //b가 a에 포함 
      } else if(idx == b.length){
        return 1;
       //한 단어의 끝에 닫기전에 다른 스펠링 
      } else {
        return a.charCodeAt(idx) - b.charCodeAt(idx);
      }
    }
  });
  for(let i = 0; i < N;){
    let j = i + 1;
    let count = 1;
    for(;j < N; j ++){
      if(a[j] === a[i]){
        count ++;
      } else {
        break;
      }
    }
    console.log(`${a[i]} ${count}`);
    i = j;
  }
}

const solve = function(){
  input();
  pro();
}

solve();

File: src/FASTCAMPUS/ALGO/D26/P20291.java
package FASTCAMPUS.ALGO.D26;
import java.io.*;
import java.util.*;

public class P20291 {
    static int N;
    static ArrayList<String> aList;
    static ArrayList<String> name;
    static Hashtable<String, Integer> count;
    static StringBuilder sb;
    static void input() {
        FastReader scan = new FastReader();
        N = scan.nextInt();
        aList = new ArrayList<String>();
        name = new ArrayList<String>();
        count = new Hashtable<String, Integer>();
        for(int i = 0; i < N; i ++){
            aList.add(scan.nextLine());
        }
        sb = new StringBuilder();
    }
    public static void pro(){
   
     for (String value : aList) {
        String extention = value.split("\\.")[1];
        name.add(extention);
     }
     Collections.sort(name);
     for(int i = 0; i < N; ){
         int count = 1;
         int j = i + 1;
         for(; j < N; j ++){
            if(name.get(i).compareTo(name.get(j)) == 0){
                count ++;
            } else {
                break;
            }
         }
         sb.append(name.get(i)).append(' ').append(count).append('\n');
         i = j;
     }
     System.out.println(sb);
    }
    public static void main(String[] args) {
        input();
        pro();
    }
    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        public FastReader(String s) throws FileNotFoundException {
            br = new BufferedReader(new FileReader(new File(s)));
        }

        String next() {
            while (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }

        long nextLong() {
            return Long.parseLong(next());
        }

        double nextDouble() {
            return Double.parseDouble(next());
        }

        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}